buildSrc/src/main/groovy/dwh-migration-dumper.java-common-conventions.gradle

```diff
@@ -73,12 +69,7 @@ dependencies {
         implementation "com.zaxxer:HikariCP:3.2.0"
         implementation "com.swrve:rate-limited-logger:2.0.0"
 
-        // TODO: when we can upgrade google-cloud-bigquery to v2+ (involves
-        // addressing breaking changes), remove the following forced versions
-        // from resolutionStrategy above:
-        // - gson
-        // - protobuf-java
-        implementation "com.google.cloud:google-cloud-bigquery:1.137.2"
+        implementation "com.google.cloud:google-cloud-bigquery:2.23.2"
```

Comment:
This change is responsible for the 4 test failures: `java.lang.NoSuchMethodError: 'com.fasterxml.jackson.core.io.ContentReference com.fasterxml.jackson.dataformat.yaml.YAMLFactory._createContentReference(java.lang.Object)'`

These failures are reproducible by running `./gradlew test` locally. What's interesting is that these failures only occur in test; if you run the dumper and dump e.g. postgresql metadata, the YAML metadata file will be written out successfully.

The method mentioned in the NSME above was introduced in Jackon v2.13. The method itself is defined in `JsonFactory` of `jackson-core`, from which `YAMLFactory` in `jackson-dataformat-yaml` extends.

I checked the dependency tree (below) and verified that there are no conflicting Jackson versions; everything is up-versioned to v2.15.0-rc1 as expected. I tried a few different Jackson versions, they all resulted in the same problem.

The only thing I can think of is that this is a Gradle bug, and that an upgrade to Gradle 7 would eliminate this. But I don't have anything to argue that case; maybe there is a plugin or Gradle library bringing a different Jackson version onto the classpath, but I can't find any way to verify that. `./gradlew buildEnvironment` doesn't show anything concerning.

I attempted the upgrade to Gradle v7 but it is non-trivial, I'm hoping to avoid it if anyone knows of any other possible causes for this.

```
mjquinn@mjquinn:~/dwh-migration-tools$ ./gradlew :dumper:app:dependencies | grep jackson
+--- com.fasterxml.jackson.core:jackson-databind -> 2.15.0-rc1
|    +--- com.fasterxml.jackson.core:jackson-annotations:2.15.0-rc1
|    |    \--- com.fasterxml.jackson:jackson-bom:2.15.0-rc1
|    |         +--- com.fasterxml.jackson.core:jackson-annotations:2.15.0-rc1 (c)
|    |         +--- com.fasterxml.jackson.core:jackson-core:2.15.0-rc1 (c)
|    |         +--- com.fasterxml.jackson.core:jackson-databind:2.15.0-rc1 (c)
|    |         +--- com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:2.15.0-rc1 (c)
|    |         \--- com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.15.0-rc1 (c)
|    +--- com.fasterxml.jackson.core:jackson-core:2.15.0-rc1
|    \--- com.fasterxml.jackson:jackson-bom:2.15.0-rc1 (*)
+--- com.fasterxml.jackson.dataformat:jackson-dataformat-yaml -> 2.15.0-rc1
|    +--- com.fasterxml.jackson.core:jackson-databind:2.15.0-rc1 (*)
|    +--- com.fasterxml.jackson.core:jackson-core:2.15.0-rc1
|    \--- com.fasterxml.jackson:jackson-bom:2.15.0-rc1 (*)
|    +--- com.fasterxml.jackson.core:jackson-core:2.14.2 -> 2.15.0-rc1
|    +--- com.fasterxml.jackson.core:jackson-annotations:2.14.2 -> 2.15.0-rc1 (*)
|    +--- com.fasterxml.jackson.core:jackson-databind:2.14.2 -> 2.15.0-rc1 (*)
|    +--- com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.14.2 -> 2.15.0-rc1
|    |    \--- com.fasterxml.jackson:jackson-bom:2.15.0-rc1 (*)
+--- com.fasterxml.jackson.core:jackson-databind:2.15.0-rc1 (c)
+--- com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:2.15.0-rc1 (c)
+--- com.fasterxml.jackson.core:jackson-databind (n)
+--- com.fasterxml.jackson.dataformat:jackson-dataformat-yaml (n)
|    +--- com.fasterxml.jackson.core:jackson-databind:2.15.0-rc1 (c)
|    +--- com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:2.15.0-rc1 (c)
|    +--- com.fasterxml.jackson.core:jackson-databind -> 2.15.0-rc1
|    |    +--- com.fasterxml.jackson.core:jackson-annotations:2.15.0-rc1
|    |    |    \--- com.fasterxml.jackson:jackson-bom:2.15.0-rc1
|    |    |         +--- com.fasterxml.jackson.core:jackson-annotations:2.15.0-rc1 (c)
|    |    |         +--- com.fasterxml.jackson.core:jackson-core:2.15.0-rc1 (c)
|    |    |         +--- com.fasterxml.jackson.core:jackson-databind:2.15.0-rc1 (c)
|    |    |         +--- com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:2.15.0-rc1 (c)
|    |    |         \--- com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.15.0-rc1 (c)
|    |    +--- com.fasterxml.jackson.core:jackson-core:2.15.0-rc1
|    |    \--- com.fasterxml.jackson:jackson-bom:2.15.0-rc1 (*)
|    +--- com.fasterxml.jackson.dataformat:jackson-dataformat-yaml -> 2.15.0-rc1
|    |    +--- com.fasterxml.jackson.core:jackson-databind:2.15.0-rc1 (*)
|    |    +--- com.fasterxml.jackson.core:jackson-core:2.15.0-rc1
|    |    \--- com.fasterxml.jackson:jackson-bom:2.15.0-rc1 (*)
|    +--- com.fasterxml.jackson.core:jackson-databind:2.15.0-rc1 (c)
|    +--- com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:2.15.0-rc1 (c)
|    |    +--- com.fasterxml.jackson.core:jackson-core:2.14.2 -> 2.15.0-rc1
|    |    +--- com.fasterxml.jackson.core:jackson-annotations:2.14.2 -> 2.15.0-rc1 (*)
|    |    +--- com.fasterxml.jackson.core:jackson-databind:2.14.2 -> 2.15.0-rc1 (*)
|    |    +--- com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.14.2 -> 2.15.0-rc1
|    |    |    \--- com.fasterxml.jackson:jackson-bom:2.15.0-rc1 (*)
|    +--- com.fasterxml.jackson.core:jackson-databind:2.15.0-rc1 (c)
|    +--- com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:2.15.0-rc1 (c)
|    +--- com.fasterxml.jackson.core:jackson-databind:2.15.0-rc1 (c)
|    +--- com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:2.15.0-rc1 (c)
+--- com.fasterxml.jackson.core:jackson-databind -> 2.15.0-rc1 (*)
+--- com.fasterxml.jackson.dataformat:jackson-dataformat-yaml -> 2.15.0-rc1 (*)
+--- com.fasterxml.jackson.core:jackson-databind:2.15.0-rc1 (c)
+--- com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:2.15.0-rc1 (c)
+--- com.fasterxml.jackson.core:jackson-databind -> 2.15.0-rc1
|    +--- com.fasterxml.jackson.core:jackson-annotations:2.15.0-rc1
|    |    \--- com.fasterxml.jackson:jackson-bom:2.15.0-rc1
|    |         +--- com.fasterxml.jackson.core:jackson-annotations:2.15.0-rc1 (c)
|    |         +--- com.fasterxml.jackson.core:jackson-core:2.15.0-rc1 (c)
|    |         +--- com.fasterxml.jackson.core:jackson-databind:2.15.0-rc1 (c)
|    |         +--- com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:2.15.0-rc1 (c)
|    |         \--- com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.15.0-rc1 (c)
|    +--- com.fasterxml.jackson.core:jackson-core:2.15.0-rc1
|    \--- com.fasterxml.jackson:jackson-bom:2.15.0-rc1 (*)
+--- com.fasterxml.jackson.dataformat:jackson-dataformat-yaml -> 2.15.0-rc1
|    +--- com.fasterxml.jackson.core:jackson-databind:2.15.0-rc1 (*)
|    +--- com.fasterxml.jackson.core:jackson-core:2.15.0-rc1
|    \--- com.fasterxml.jackson:jackson-bom:2.15.0-rc1 (*)
|    +--- com.fasterxml.jackson.core:jackson-core:2.14.2 -> 2.15.0-rc1
|    +--- com.fasterxml.jackson.core:jackson-annotations:2.14.2 -> 2.15.0-rc1 (*)
|    +--- com.fasterxml.jackson.core:jackson-databind:2.14.2 -> 2.15.0-rc1 (*)
|    +--- com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.14.2 -> 2.15.0-rc1
|    |    \--- com.fasterxml.jackson:jackson-bom:2.15.0-rc1 (*)
+--- com.fasterxml.jackson.core:jackson-databind:2.15.0-rc1 (c)
+--- com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:2.15.0-rc1 (c)
|    +--- com.fasterxml.jackson.core:jackson-databind:2.15.0-rc1 (c)
|    +--- com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:2.15.0-rc1 (c)
|    +--- com.fasterxml.jackson.core:jackson-databind -> 2.15.0-rc1
|    |    +--- com.fasterxml.jackson.core:jackson-annotations:2.15.0-rc1
|    |    |    \--- com.fasterxml.jackson:jackson-bom:2.15.0-rc1
|    |    |         +--- com.fasterxml.jackson.core:jackson-annotations:2.15.0-rc1 (c)
|    |    |         +--- com.fasterxml.jackson.core:jackson-core:2.15.0-rc1 (c)
|    |    |         +--- com.fasterxml.jackson.core:jackson-databind:2.15.0-rc1 (c)
|    |    |         +--- com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:2.15.0-rc1 (c)
|    |    |         \--- com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.15.0-rc1 (c)
|    |    +--- com.fasterxml.jackson.core:jackson-core:2.15.0-rc1
|    |    \--- com.fasterxml.jackson:jackson-bom:2.15.0-rc1 (*)
|    +--- com.fasterxml.jackson.dataformat:jackson-dataformat-yaml -> 2.15.0-rc1
|    |    +--- com.fasterxml.jackson.core:jackson-databind:2.15.0-rc1 (*)
|    |    +--- com.fasterxml.jackson.core:jackson-core:2.15.0-rc1
|    |    \--- com.fasterxml.jackson:jackson-bom:2.15.0-rc1 (*)
|    +--- com.fasterxml.jackson.core:jackson-databind:2.15.0-rc1 (c)
|    +--- com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:2.15.0-rc1 (c)
|    |    +--- com.fasterxml.jackson.core:jackson-core:2.14.2 -> 2.15.0-rc1
|    |    +--- com.fasterxml.jackson.core:jackson-annotations:2.14.2 -> 2.15.0-rc1 (*)
|    |    +--- com.fasterxml.jackson.core:jackson-databind:2.14.2 -> 2.15.0-rc1 (*)
|    |    +--- com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.14.2 -> 2.15.0-rc1
|    |    |    \--- com.fasterxml.jackson:jackson-bom:2.15.0-rc1 (*)
|    +--- com.fasterxml.jackson.core:jackson-databind:2.15.0-rc1 (c)
|    +--- com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:2.15.0-rc1 (c)
|    +--- com.fasterxml.jackson.core:jackson-databind:2.15.0-rc1 (c)
|    +--- com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:2.15.0-rc1 (c)
+--- com.fasterxml.jackson.core:jackson-databind -> 2.15.0-rc1 (*)
+--- com.fasterxml.jackson.dataformat:jackson-dataformat-yaml -> 2.15.0-rc1 (*)
+--- com.fasterxml.jackson.core:jackson-databind:2.15.0-rc1 (c)
+--- com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:2.15.0-rc1 (c)
|    +--- com.fasterxml.jackson.core:jackson-databind:2.15.0-rc1 (c)
|    +--- com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:2.15.0-rc1 (c)
|    +--- com.fasterxml.jackson.core:jackson-databind -> 2.15.0-rc1
|    |    +--- com.fasterxml.jackson.core:jackson-annotations:2.15.0-rc1
|    |    |    \--- com.fasterxml.jackson:jackson-bom:2.15.0-rc1
|    |    |         +--- com.fasterxml.jackson.core:jackson-annotations:2.15.0-rc1 (c)
|    |    |         +--- com.fasterxml.jackson.core:jackson-core:2.15.0-rc1 (c)
|    |    |         +--- com.fasterxml.jackson.core:jackson-databind:2.15.0-rc1 (c)
|    |    |         +--- com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:2.15.0-rc1 (c)
|    |    |         \--- com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.15.0-rc1 (c)
|    |    +--- com.fasterxml.jackson.core:jackson-core:2.15.0-rc1
|    |    \--- com.fasterxml.jackson:jackson-bom:2.15.0-rc1 (*)
|    +--- com.fasterxml.jackson.dataformat:jackson-dataformat-yaml -> 2.15.0-rc1
|    |    +--- com.fasterxml.jackson.core:jackson-databind:2.15.0-rc1 (*)
|    |    +--- com.fasterxml.jackson.core:jackson-core:2.15.0-rc1
|    |    \--- com.fasterxml.jackson:jackson-bom:2.15.0-rc1 (*)
|    +--- com.fasterxml.jackson.core:jackson-databind:2.15.0-rc1 (c)
|    +--- com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:2.15.0-rc1 (c)
|    |    +--- com.fasterxml.jackson.core:jackson-core:2.14.2 -> 2.15.0-rc1
|    |    +--- com.fasterxml.jackson.core:jackson-annotations:2.14.2 -> 2.15.0-rc1 (*)
|    |    +--- com.fasterxml.jackson.core:jackson-databind:2.14.2 -> 2.15.0-rc1 (*)
|    |    +--- com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.14.2 -> 2.15.0-rc1
|    |    |    \--- com.fasterxml.jackson:jackson-bom:2.15.0-rc1 (*)
|    +--- com.fasterxml.jackson.core:jackson-databind:2.15.0-rc1 (c)
|    +--- com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:2.15.0-rc1 (c)
|    +--- com.fasterxml.jackson.core:jackson-databind:2.15.0-rc1 (c)
|    +--- com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:2.15.0-rc1 (c)
+--- com.fasterxml.jackson.core:jackson-databind -> 2.15.0-rc1 (*)
+--- com.fasterxml.jackson.dataformat:jackson-dataformat-yaml -> 2.15.0-rc1 (*)
+--- com.fasterxml.jackson.core:jackson-databind:2.15.0-rc1 (c)
+--- com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:2.15.0-rc1 (c)
```

Reply:
Resolved by shadowing `com.fasterxml` in `lib-ext-hive-metastore`; this was confirmed to be the issue by noting in the JDiagnostic result below that `ObjectMapper` was sourced from `lib-ext-hive-metastore` as the first classpath entry taking precedence over the desired `jackson-databind` entry.